### ISSUE TITLE
Toggle Mission steps in Future Missions

### ISSUE THREAD

**Issue by rnyoakum** [raceintospace/raceintospace#196]
The Future Missions screen has the useful feature of being able to display the flight path taken in a planned mission, as well as marking the steps which compose it. At the moment, when a mission is selected the user may click on the button (between the scroll arrows) or press the space bar, activating the mission steps mode for the currently previewed mission. Navigating to another mission deactivates this mode. This got me wondering:
I) Is this supposed to be a toggled option? What was the original BARIS behavior?
II) If not the original behavior, should we add a configuration setting to make it a toggled option?

**Comment by peyre** [comment]
No, the original BARIS/RIS behavior, and what ours should have, is that navigating to another mission and then hitting the button will show the steps for the currently selected mission.

    On Friday, November 16, 2018, 5:11:23 PM PST, Ryan Yoakum <notifications@github.com> wrote:  
 
 
The Future Missions screen has the useful feature of being able to display the flight path taken in a planned mission, as well as marking the steps which compose it. At the moment, when a mission is selected the user may click on the button (between the scroll arrows) or press the space bar, activating the mission steps mode for the currently previewed mission. Navigating to another mission deactivates this mode. This got me wondering:
I) Is this supposed to be a toggled option? What was the original BARIS behavior?
II) If not the original behavior, should we add a configuration setting to make it a toggled option?

**Comment by peyre** [comment]
So what we have now is what I've always expected to see, at least.  I don't see why we couldn't add a config option for it though.

**Comment by rnyoakum** [comment]
One idea I had recently was to have an option whereby the flight path would be animated, drawing the mission step by step. But that looks harder to implement than I'd hoped.

**Comment by peyre** [comment]
Oh, I can see that being difficult to pull off.  Right now I'm tinkering with future.cpp myself, as it happens--I'm trying to construct a popup of requirement penalties when you click on the A-OK/WARNING/DANGER button.
